src/useFormState.test.tsx

```diff
@@ -628,6 +631,40 @@ describe("useFormState", () => {
     // Then the form is marked as not loading
     expect(r.loading()).toHaveTextContent("false");
   });
+
+  it("updates AutoSaveStatusProvider when autoSaving", async () => {
+    type FormValue = Pick<AuthorInput, "firstName">;
+    const config: ObjectConfig<FormValue> = { firstName: { type: "value" } };
+    const { result } = renderHook(
+      () => {
+        const autoSave = useAutoSaveStatus();
+        const fs = useFormState({ config, autoSave: async () => {} });
+
+        return { fs, autoSave };
+      },
+      {
+        wrapper: ({ children }) => (
+          <AutoSaveStatusProvider>
+            <Observer>{() => <>{children}</>}</Observer>
+          </AutoSaveStatusProvider>
+        ),
+      },
+    );
+    expect(result.current.autoSave.status).toBe(AutoSaveStatus.IDLE);
+
+    // When we invoke a change and autoSave fires
+    actOnHook(() => {
+      result.current.fs.set({ firstName: "test" });
+    });
+    // Idk where/how to intercept "Saving" but once it settles...
```

Comment:
I'm kind of surprised you can't assert for `expect(result.current.autoSave.status).toBe(AutoSaveStatus.SAVING)`? Especially if you have to call `runAllTimers` in order to get the "done" status.

Reply:
😕 Because of the implementation of `useFormState/maybAutoSave/setTimeout(maybeAutoSave,0)`, autoSave doesn't run immediately so I can't step in the middle of it, and by the time jest.runAllTimesr() executes, autoSave has fired and resolved in an instant so it's already done.

There must be _some_ way to interject but I only have so many hours in a day 😅

Reply:
It's also kinda weird because this **first one works**, the second one doesn't:

```jsx
await actOnHook(async () => {
  jest.runAllTimers();
});

actOnHook(() => {
  jest.runAllTimers();
});
```